full-stack-tests/presentation/src/frontend/UnifiedSelection.test.ts

```diff
@@ -185,6 +188,172 @@ describe("Unified Selection", () => {
+        return builder.insertElement(spatialCategoryProps);
+      };
+
+      const addPhysicalObject = (builder: TestIModelBuilder, modelId: string, categoryId: string, elemCode = Code.createEmpty()) => {
+
+        const physicalObjectProps: PhysicalElementProps = {
+          classFullName: "Generic:PhysicalObject",
+          model: modelId,
+          category: categoryId,
+          code: elemCode,
+        };
+        return builder.insertElement(physicalObjectProps);
+      };
+
+      const objectIdArray = Array<Id64String>();
+      let groupingNodesIModel: IModelConnection;
+      beforeEach(async () => {
+        groupingNodesIModel = await buildTestIModel("GroupByClass", (builder) => {
+          const physicalPartitionId = addPartition("BisCore:PhysicalPartition", builder, "TestDrawingModel");
+          const definitionPartitionId = addPartition("BisCore:DefinitionPartition", builder, "TestDefinitionModel");
+          const physicalModelId = addModel("BisCore:PhysicalModel", builder, physicalPartitionId);
+          const definitionModelId = addModel("BisCore:DefinitionModel", builder, definitionPartitionId);
+          const categoryId = addSpatialCategory(builder, definitionModelId, "Test SpatialCategory");
+          objectIdArray.push(addPhysicalObject(builder, physicalModelId, categoryId, new Code({ value: "code", scope: "0x1", spec: "0x1" })));
+          objectIdArray.push(addPhysicalObject(builder, physicalModelId, categoryId, new Code({ value: "code", scope: "0x1", spec: "0x2" })));
+          objectIdArray.push(addPhysicalObject(builder, physicalModelId, categoryId, new Code({ value: "code2", scope: "0x1", spec: "0x2" })));
+        });
+
+        Presentation.selection.clearSelection("", groupingNodesIModel);
+        handler = new ViewportSelectionHandler({ imodel: groupingNodesIModel });
+      });
+
+      afterEach(async () => {
+        await Presentation.presentation.rulesets().clear();
+        objectIdArray.splice(0);
+        await groupingNodesIModel.close();
+        handler.dispose();
+      });
+
+      it("hilites grouping nodes grouped by class", async () => {
+        const groupByClassRuleset: Ruleset = {
+          id: "groupByClassRuleset",
+          rules: [{
+            ruleType: RuleTypes.RootNodes,
+            specifications: [{
+              specType: ChildNodeSpecificationTypes.InstanceNodesOfSpecificClasses,
+              classes: { schemaName: "BisCore", classNames: [GeometricElement3d.className] },
+              arePolymorphic: true,
+              instanceFilter: "this.Parent = NULL",
+              groupByClass: true,
+              groupByLabel: false,
+            }],
+          }],
+        };
+
+        await using<RegisteredRuleset, Promise<void>>(await Presentation.presentation.rulesets().add(groupByClassRuleset), async () => {
+          const rootNodes = await Presentation.presentation.getNodes({ imodel: groupingNodesIModel, rulesetOrId: groupByClassRuleset.id });
```

Comment:
you don't need to call `Presentation.presentation.rulesets().add(groupByClassRuleset)` if you pass ruleset instead of its id to the `getNodes` call

Reply:
Passed in ruleset and removed `using`.